src/coreclr/jit/codegenarm.cpp

```diff
@@ -119,12 +119,12 @@ BasicBlock* CodeGen::genCallFinally(BasicBlock* block)
 {
     assert(block->KindIs(BBJ_CALLFINALLY));
 
-    GetEmitter()->emitIns_J(INS_bl, block->GetTarget());
-
     BasicBlock* nextBlock = block->Next();
 
     if (block->HasFlag(BBF_RETLESS_CALL))
     {
+        GetEmitter()->emitIns_J(INS_bl, block->GetTarget());
```

Comment:
Unlike other uses of NoGC regions the use in `genCallFinally` was starting the NoGC region on instruction that already cannot do GC. 

We need to agree on the same semantics for all the uses, so `genCallFinally` had to be adjusted to work the same as in other uses of NoGC like block copying, tailcall arg setups,...

Reply:
We could have it this way (and adjust uses in block copying, etc..), we just need to have it the same way in all cases. 

However, starting NoGC region on an instruction group that _will_ make GC unsafe seems a lot more convenient as we may want to start NoGC region before we know exact instructions that will go into it.